src/modules/roc_sndio/target_sox/roc_sndio/player.cpp

```diff
@@ -71,18 +65,49 @@ bool Player::open(const char* name, const char* type) {
         return false;
     }
 
+    size_t in_rate = size_t(out_signal_.rate);
+    size_t out_rate = size_t(output_->signal.rate);
+
+    if (in_rate != 0 && in_rate != out_rate) {
+        roc_log(LogError, "player: sample rates mismatch, auto-detected=%zu, input=%zu",
```

Comment:
I think the error message should be something like:

"can't open output file or device with the required sample rate: required=%lu suggested=%lu".

Also some notes:

* we don't use %z, it's not portable
* we don't use commas when logging multiple values, they don't increase readability, but they make grepping a bit harder

Reply:
Fixed